content/concepts/did-ddo.md

```diff
@@ -1,173 +1,714 @@
+      "language": "Node.js",
+      "version": "1.0.0",
+      "container": {
+        "entrypoint": "node $ALGO",
+        "image": "ubuntu",
+        "tag": "latest",
+        "checksum": "44e10daa6637893f4276bb8d7301eb35306ece50f61ca34dcab550"
+      }
+    }
+  }
+}
+```
 
-Asset metadata must be included as one of the objects inside the `"service"` array, with type `"metadata"`.
+### Services
 
-## DDO Service Types
+Services define the access for an asset, and each service is represented by its respective datatoken.
 
-There are many possible service types for a DDO.
+An asset should have at least one service to be actually accessible, and can have as many services which make sense for a specific use case.
 
-- `metadata` - describing the asset
-- `access` - describing how the asset can be downloaded
-- `compute` - describing how the asset can be computed upon
+| Attribute              | Type                        | Required                        | Description                                                                                                                                  |
+| ---------------------- | --------------------------- | ------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------- |
+| **`id`**               | `string`                    | **✓**                           | Unique ID                                                                                                                                    |
+| **`type`**             | `string`                    | **✓**                           | Type of service (`access`, `compute`, `wss`, etc.                                                                                            |
+| **`name`**             | `string`                    |                                 | Service friendly name                                                                                                                        |
+| **`description`**      | `string`                    |                                 | Service description                                                                                                                          |
+| **`datatokenAddress`** | `string`                    | **✓**                           | Datatoken address                                                                                                                            |
+| **`serviceEndpoint`**  | `string`                    | **✓**                           | Provider URL (schema + host)                                                                                                                 |
+| **`files`**            | [Files](#files)             | **✓**                           | Encrypted file URLs.                                                                                                                         |
```

Comment:
After actually implementing stuff, `files` should be `string` and the `Files` object should/will be used only when interacting with `Provider`.